featurebyte/session/snowflake.py

```diff
@@ -186,51 +230,25 @@ def fetch_query_result_impl(self, cursor: Any) -> pd.DataFrame | None:
                 return super().fetch_query_result_impl(cursor)
         return None
 
-    @staticmethod
-    def _get_column_name_to_db_var_type(cursor: Any) -> dict[str, DBVarType]:
-        """
-        Get DB var type of data being fetched from the snowflake cursor.
-
-        Parameters
-        ----------
-        cursor: Any
-            The connection cursor
-
-        Returns
-        -------
-        dict[str, DBVarType]
-            A dictionary mapping column names to DB var types
-        """
-        output = {}
-        # See https://docs.snowflake.com/en/developer-guide/python-connector/python-connector-api#label-python-connector-type-codes
-        # for more details. We currently only map one value since that's the only one we need to do any special
-        # handling for. If we need to apply more special handling in the future, we can add more mappings here.
-        type_code_to_db_var_type_mapping: dict[int, DBVarType] = {10: DBVarType.ARRAY}
-        for metadata in cursor.description:
-            if not isinstance(metadata, ResultMetadata):
-                continue
-            type_code = metadata.type_code
-            if type_code in type_code_to_db_var_type_mapping:
-                output[metadata.name] = type_code_to_db_var_type_mapping[type_code]
-        return output
-
     async def fetch_query_stream_impl(self, cursor: Any) -> AsyncGenerator[pa.RecordBatch, None]:
-        col_name_to_db_var_type = self._get_column_name_to_db_var_type(cursor)
         # fetch results in batches and write to the stream
+        schema = None
         try:
-            counter = 0
+            schema = self._get_schema_from_cursor(cursor)
             for table in cursor.fetch_arrow_batches():
-                for batch in pa_table_to_record_batches(table, col_name_to_db_var_type):
-                    counter += 1
+                for batch in table.cast(schema).to_batches():
                     yield batch
-            if counter == 0:
-                # Arrow batch is empty, need to return empty table with schema
-                table = cursor.get_result_batches()[0].to_arrow()
-                yield pa_table_to_record_batches(table)[0]
-        except NotSupportedError:
+            # return empty table to ensure correct schema is returned
+            yield pa.record_batch(
+                pd.DataFrame(columns=[field.name for field in schema]), schema=schema
+            )
+        except (NotSupportedError, AttributeError):
```

Comment:
Where is AttributeError expected to be raised? Would be good to leave a comment on that since it's not obvious. If the exception type is something we can control, it would be nice to raise a more specific exception here since AttributeError is rather generic.